### PR TITLE
chore: add automation for @nrwl + nx package updates

### PR DESCRIPTION
**Pull request by JamesHenry**
I have been using this approach very successfully for a while on angular-eslint

E.g. https://github.com/angular-eslint/angular-eslint/pull/1135

Nx has its own powerful built in migration tool `nx migrate`, which will update relevant packages (therefore those are added to ignore in renovate config) and even run required code and config migrations (codemods) if applicable.

Here you can see an example of such automatic config updates: https://github.com/typescript-eslint/typescript-eslint/pull/5798

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "lint-markdown-fix": "yarn lint-markdown --fix",
     "lint-markdown": "markdownlint \"**/*.md\" --config=.markdownlint.json --ignore-path=.markdownlintignore",
     "lint": "cross-env NODE_OPTIONS=\"--max-old-space-size=16384\" eslint .",
-    "postinstall": "yarn patch-package && yarn husky install && yarn build",
+    "postinstall": "tools/scripts/postinstall.sh",
```
Comment (bradzacher) [inline package.json:41]: Using a shell script isn't great because it isn't portable to Windows.

Is there another way we can wire this up?

Comment (JamesHenry) [reply package.json:41]: Ah, damn. Open to any suggestions, I've just found that having the ability to ensure postinstall steps aren't run when automated installs happen as part of migrations helps make sure they always works. So this simple env var + shell script approach was what I settled on at the time

Comment (JamesHenry) [reply package.json:41]: ts-node + child process is one option?

File: .github/renovate.json [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+  "ignoreDeps": [
+    "@nrwl/cli",
+    "@nrwl/devkit",
+    "@nrwl/jest",
+    "@nrwl/nx-cloud",
+    "@nrwl/tao",
+    "@types/jest",
+    "cypress",
+    "jest",
+    "ts-jest"
```
Comment (bradzacher) [inline .github/renovate.json:13]: how come nx manages these deps for us?

File: package.json (hunk 1 above)
Comment (bradzacher) [reply package.json:41]: I don't know a good solution - maybe just a ts-node script is the best option? ugh. would need to do some googling.

i know we do have contributors that work on windows so we'd need to solve this.

perhaps we can just do both a `.bat` and `.sh` version of the script and call the relevant one from a simple ts-node script? IDK

Comment (mrlonis) [reply package.json:41]: Could you write the script as a Node `.js` script? If so, you could leverage it in `postinstall` much like the `decorate-angular-cli.js` file works for an Nx Angular repo:

```json
{
  ...,
  "scripts": {
    ...
    "postinstall": "node ./decorate-angular-cli.js && ngcc --properties es2020 browser module main",
    ...
  },
  ...
}
```

Comment (JoshuaKGoldberg) [reply package.json:41]: https://github.com/google/zx looks pretty sweet 😎

Comment (bradzacher) [reply package.json:41]: Holy shit zx is amazing.
I wish I'd learned about it sooner!!!

File: .github/renovate.json (hunk 2 above)
Comment (JamesHenry) [reply .github/renovate.json:13]: We don't have to let it, but Nx comes with an executor for jest and so if we leverage that we can basically forget about jest upgrades and no that the Nx Team will test the collection of packages e2e on each release...

If we want to stay on the bleeding edge of jest at all times then we choose not to follow that approach.

WDYT?

Comment (JamesHenry) [reply .github/renovate.json:13]: Same goes for cypress

Comment (JamesHenry) [reply .github/renovate.json]: I kept jest and cypress out of the list for now

File: package.json (hunk 1 above)
Comment (JamesHenry) [reply package.json]: I did have a play with zx but it's ESM only and therefore requires some additional fiddling with tsconfigs and package.jsons

We already have `execa` in our dependency tree so I decided to formalize that devDependency instead and made a small wrapper `$` function which matches `zx` usage and output

File: package.json [hunk 3]
```diff
@@ -86,6 +86,7 @@
     "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-jest": "^26.1.5",
     "eslint-plugin-simple-import-sort": "^8.0.0",
+    "execa": "5.1.1",
```
Comment (JamesHenry) [inline package.json:89]: Pinned to the last non-ESM version

**Comment by JamesHenry** [comment]
Netlify redirect rules check seems stuck...

**Review by bradzacher — APPROVED** [review]

**Comment by bradzacher** [comment]
whatugot

**Pull request merged**